### PR TITLE
Add proposal of ci improvement 

### PR DESCRIPTION
**Pull request by wy65701436**

### CODE REVIEW

File: proposals/new/ciimprovement.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## Background
+
+Currently, there are three CI systems running on Harbor repo that are travis, drone 
+and jenkins. 
+
+*Travis CI executes UI test.
+*Drone CI executes E2E test and packages the installer.
+*Jenkins CI executes full tests daily, it's locates in the vmware internal network.   
+
+Drone is running in the vmware internal network, and any pull request created by
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/vmware/VMware/

File: proposals/new/ciimprovement.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Currently, there are three CI systems running on Harbor repo that are travis, drone 
+and jenkins. 
+
+*Travis CI executes UI test.
+*Drone CI executes E2E test and packages the installer.
+*Jenkins CI executes full tests daily, it's locates in the vmware internal network.   
+
+Drone is running in the vmware internal network, and any pull request created by 
+contributor who is not a memeber of vmware community is not allowed to be executed
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/vmware/VMware/

File: proposals/new/ciimprovement.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+automaticlly. This is not friendly for the commumity contributor.
+
+Jenkins is running in the vmware internal network and infrastructure, all of the E2E
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/vmware/VMware/

File: proposals/new/ciimprovement.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+tests - almost as UI based - are executed daily. The results is published to the 
+internal slack channel. This is not friendly for the communtion contributor, anyone
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/slack/Slack/

File: proposals/new/ciimprovement.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+who does not have permission to access vmware network has no way to view the results.
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/vmware/VMware/

File: proposals/new/ciimprovement.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+## Proposal
+Unify the CI systems, make it easy to the community contributor to triage their
+code changes, know the CI results and develop new tests.
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/know the CI results/have access to the CI results/

File: proposals/new/ciimprovement.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+1, Deprecate drone CI to run tests, leave it as build system only. Any pull request
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: In the long-term, the community will need to have access to the build system. Perhaps we should consider this particular ask in another proposal when the need becomes pressing. Just wanted to put this out there, though.

File: proposals/new/ciimprovement.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+will not trigger a drone job, only push(code merge) will trigger a drone job to package
+installer and push it to the google storage. Drone will be an internal build system, 
+the package URL is public, anyone could get the latest dev/release build with the
+public URL.
+
+2, Enhance travis CI to run jobs paraelle, it's in charge of testing execution, include
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/paraelle/parallel/

File: proposals/new/ciimprovement.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+UT/API. Any pull request will trigger four CI jobs, that are UT/API on DB auth/API on 
+LDAP auth/Package installer, all of these results are associate with commit.
+
+3, Public the nightly results generated by internal jenkins CI systems, anyone could
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: As mentioned above, this step will ultimately need to be accessible to contributors / non-VMware maintainers. Perhaps we should leave this as another step / proposal as needed.

File: proposals/new/ciimprovement.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+installer and push it to the google storage. Drone will be an internal build system, 
+the package URL is public, anyone could get the latest dev/release build with the
+public URL.
+
+2, Enhance travis CI to run jobs paraelle, it's in charge of testing execution, include
+UT/API. Any pull request will trigger four CI jobs, that are UT/API on DB auth/API on 
+LDAP auth/Package installer, all of these results are associate with commit.
+
+3, Public the nightly results generated by internal jenkins CI systems, anyone could 
+get the status from repo badge to see testing results and to konw whether their code
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/konw/know/

File: proposals/new/ciimprovement.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+2, Enhance travis CI to run jobs paraelle, it's in charge of testing execution, include
+UT/API. Any pull request will trigger four CI jobs, that are UT/API on DB auth/API on 
+LDAP auth/Package installer, all of these results are associate with commit.
+
+3, Public the nightly results generated by internal jenkins CI systems, anyone could 
+get the status from repo badge to see testing results and to konw whether their code 
+introduces regression.
+
+4, Define a API development process, any new contributor could follow the guidline to
```
Comment (clouderati) [inline proposals/new/ciimprovement.md]: s/guidline/guideline/

**Comment by clouderati** [comment]
Hi @wy65701436 – can you please sign-off on your commit so the DCO bot is happy?

**Comment by clouderati** [comment]
Hi @wy65701436 – any updates on this? Thanks!

File: proposals/new/ciimprovement.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+As for donate to CNCF, harbor has to enhance it's CI system to make it easier for
+the contributor to commit code and write tests, either UI or API tests. Provide an 
+single CI system is easy to maintain and easy to learn.
+
+## Background
+
+Currently, there are three CI systems running on Harbor repo that are travis, drone 
+and jenkins. 
+
+*Travis CI executes UI test.
```
Comment (steven-zou) [inline proposals/new/ciimprovement.md]: Please correct the format

**Pull request closed without merging**